gdal/frmts/dimap/dimapdataset.cpp

```diff
@@ -1301,7 +1305,15 @@ int DIMAPDataset::ReadImageInformation2()
         if( poImageDS->GetGeoTransform(adfGeoTransform) == CE_None &&
             !(adfGeoTransform[0] <= 1.5 && fabs(adfGeoTransform[3]) <= 1.5) )
         {
-            bHaveGeoTransform = TRUE;
+            if ( nImageDSCol == 1 && nImageDSRow == 1 ) {
+                bHaveGeoTransform = TRUE;
+            } else if ( adfGeoTransform[2] == 0 && adfGeoTransform[4] == 0 ) {
+                //if north-up images, fix up the origin if we did not get the
+                //geotransform from the top-left tile
+                adfGeoTransform[0] -= (nImageDSCol-1) * adfGeoTransform[1] * nTileWidth;
```

Comment:
Could probably be generalized with:
```suggestion
                adfGeoTransform[0] -= (nImageDSCol-1) * adfGeoTransform[1] * nTileWidth + (nImageDSRow-1) * adfGeoTransform[2] * nTileHeight;
```

Reply:
and just remove the if()  + else if()

Reply:
Thanks for the suggestion, I was too lazy to find what needed to be done in case of rotation. @rouault you say "probably", does that mean this won't work all the time ?

Reply:
> you say "probably", does that mean this won't work all the time ?

I'm reasonably confident this should work (this is the standard formula), but didn't test of course :-)

Reply:
any error would only concern dimaps that didn't load previously, so let's go with this :)